tests/testapp/test_models.py

```diff
@@ -236,6 +237,22 @@ def test_use_index_primary(self):
         used = used_indexes(cap.query)
         assert len(used) == 0 or "PRIMARY" in used
 
+    def test_use_index_inner_query(self):
+        title_idx = index_name(Book, "title")
+        with CaptureLastQuery() as cap:
+            list(
+                Author.objects.annotate(
+                    has_books=Exists(
+                        Book.objects.filter(author_id=OuterRef("id"), title__gt="")
+                    )
+                )
+                .filter(has_books=True)
+                .use_index(title_idx, table_name="testapp_book")
```

Comment:
why isn't this on the `Book` subquery?

Reply:
for some reason I thought it wouldn't work without table_name in the subquery since the query rewriting works in context of the finished query, but seems like that wasn't correct. will move it